enterprise/api/v1/views.py

```diff
@@ -64,6 +64,71 @@ class EnterpriseCustomerViewSet(EnterpriseReadOnlyModelViewSet):
     filter_fields = FIELDS
     ordering_fields = FIELDS
 
+    @detail_route(
+        permission_classes=[permissions.IsAuthenticated, IsServiceUserOrReadOnly],
+        authentication_classes=[JwtAuthentication, BearerAuthentication, SessionAuthentication],
+        throttle_classes=[ServiceUserThrottle],
+    )  # pylint: disable=invalid-name,unused-argument
+    def courses(self, request, pk=None):
+        """
+        Retrieve the list of courses contained within the catalog linked to this enterprise.
+
+        Only courses with active course runs are returned. A course run is considered active if it is currently
+        open for enrollment, or will open in the future.
+        ---
+        serializer: serializers.CourseSerializerExcludingClosedRuns
+        """
+        enterprise_customer = self.get_object()
+
+        # Make sure there is a catalog associated with the enterprise
+        if not enterprise_customer.catalog:
+            error_message = (
+                "No catalog is associated with Enterprise {enterprise_name} from endpoint '{path}'.".format(
+                    enterprise_name=enterprise_customer.name,
+                    path=request.get_full_path()
+                )
+            )
+            logger.error(error_message)
+            raise NotFound("The resource you are looking for does not exist: " + error_message)
+
+        # Ensure that the user making this request is either a staff user or is linked to the enterprise
+        if not request.user.is_staff and not models.EnterpriseCustomerUser.objects.filter(
+                enterprise_customer=enterprise_customer,
+                user_id=request.user.id,
+        ).exists():
+            error_message = (
+                "User '{username}' is not associated with Enterprise {enterprise_name} from endpoint '{path}'.".format(
+                    username=request.user.username,
+                    enterprise_name=enterprise_customer.name,
+                    path=request.get_full_path()
+                )
+            )
+            logger.error(error_message)
+            raise PermissionDenied("The user does not have permission to access this resource: " + error_message)
```

Comment:
I've looked into this (http://www.django-rest-framework.org/api-guide/permissions/), and my only concern about doing it that way is that we would lose expressiveness in the message returned to the user, as one message is returned for the PermissionClass and I don't see a way to dynamically generate it as we do here.